confgenerator/logging_receivers.go

```diff
@@ -53,9 +53,10 @@ func (r LoggingReceiverFiles) Components(tag string) []fluentbit.Component {
 }
 
 type LoggingReceiverFilesMixin struct {
-	IncludePaths            []string       `yaml:"include_paths,omitempty"`
-	ExcludePaths            []string       `yaml:"exclude_paths,omitempty"`
-	WildcardRefreshInterval *time.Duration `yaml:"wildcard_refresh_interval,omitempty" validate:"omitempty,min=1s,multipleof_time=1s"`
+	IncludePaths            []string        `yaml:"include_paths,omitempty"`
+	ExcludePaths            []string        `yaml:"exclude_paths,omitempty"`
+	WildcardRefreshInterval *time.Duration  `yaml:"wildcard_refresh_interval,omitempty" validate:"omitempty,min=1s,multipleof_time=1s"`
+	MultilineRules          []MultilineRule `yaml:"-"`
```

Comment:
Just double checking, we do not want to expose a `multiline_rules` yaml configuration parameter to the end user in receivers like `files`, `tcp` or `elasticsearch`, which I believe all use `LoggingReceiverFilesMixin`. In other words, it should be an implementation details at fluent bit config generater side, instead of a user visible configuration.

Can we confirm this is the case, and if not, remove this field from the publicly visible side?

Reply:
https://pkg.go.dev/github.com/goccy/go-yaml@v1.9.4#Marshal - ` if the key is "-", the field is ignored.` 

So, the `MultlineRules` field is ignored when marshalling/unmarshalling user configs, and can only be set programmatically because of the tag. 

I added an invalid config unit test specifying the field, so we can confirm that that is the case. That test fails if the yaml tag is removed.

Reply:
Sounds great!

Reply:
Thanks @BinaryFissionGames for fixing the things right on using the parser within the Input plugin:) 😉

FYI, I am working on enabling multiline parser on tail input. After looking at the existing code, i plan to work on my own space (meaning reusing very limited existing code). After my PR of enabling multiline parser is in(WIP), then we shall do a refactoring work on those 3d party apps which uses multiline parser.